io/net/net_tools.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+# Based on code by Hong Bo Peng <penghb@cn.ibm.com>
+# copyright: 2003, 2015 IBM Corp
+
+import os
+import re
+import avocado
+from avocado import Test
+from avocado import main
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+def install_dependencies():
+    """
+    To check and install dependencies for the test
+    """
+    detected_distro = distro.detect()
+    sm = SoftwareManager()
+    if detected_distro.name == "SuSE":
+        net_tools = ("net-tools", "traceroute")
+    else:
+        net_tools = ("net-tools", "hostname", "traceroute")
+    for pkg in net_tools:
+        if not sm.check_installed(pkg) and not sm.install(pkg):
+            raise AssertionError("%s package is need to test" % pkg)
+
+
+class Hostname(Test):
+    """
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    Test to verify the functionality of the hostname command
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.restore_hostname = False
+        # Get Hostname
+        hostname = process.system_output("hostname").strip('\n')
+        if not hostname:
+            # set hostname if not set
+            process.system("hostname localhost.localdomain", sudo=True)
+            hostname = "localhost.localdomain"
+        self.hostname = hostname
+
+    @avocado.fail_on(process.CmdError)
+    def test_hostname(self):
+        """
+        i.Verifies hostname command options and
+        ii.Test to change the hostname
+        """
+        output = process.system_output("hostname")
+        if not output:
+            self.fail("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            ret = process.run("hostname -%s" % option, ignore_status=True)
+            if ret.exit_status:
+                self.fail("Hostname reported non-zero status %s for option %s"
```

Comment:
You ought to log the full `ret` to see what was executed and what was the result including stdout/stderr, otherwise you have no hints of why it reported that exit status.